diff --git a/build-scripts/build-doc b/build-scripts/build-doc
index 077f836..18f85ee 100755
--- a/build-scripts/build-doc
+++ b/build-scripts/build-doc
@@ -10,7 +10,7 @@ pip3 install sphinx sphinx_rtd_theme
 ./configure --enable-doc-maintenance
 make -j$(nproc) doc-dist DOC_DEST=doc
 zip -r doc.zip doc
-version=$(egrep '^release' manual/conf.py | cut -d"'" -f 2)
+version=$(grep -E '^release' manual/conf.py | cut -d"'" -f 2)
 mv doc qpdf-${version}-doc
 mkdir distribution
 zip -r distribution/qpdf-${version}-doc.zip qpdf-${version}-doc
diff --git a/qpdf/qtest/qpdf/diff-encrypted b/qpdf/qtest/qpdf/diff-encrypted
index ac5428c..f6741c2 100755
--- a/qpdf/qtest/qpdf/diff-encrypted
+++ b/qpdf/qtest/qpdf/diff-encrypted
@@ -1,5 +1,5 @@
 #!/bin/sh
-lines=$(expr 0 + $(diff "$1" "$2" | egrep '^[<>]' | egrep -v '(Date|InstanceID)' | wc -l))
+lines=$(expr 0 + $(diff "$1" "$2" | grep -E '^[<>]' | grep -E -v '(Date|InstanceID)' | wc -l))
 if [ "$lines" = "0" ]; then
    echo okay
 else
diff --git a/qpdf/qtest/qpdf/diff-ignore-ID-version b/qpdf/qtest/qpdf/diff-ignore-ID-version
index 72def44..de4c7c9 100755
--- a/qpdf/qtest/qpdf/diff-ignore-ID-version
+++ b/qpdf/qtest/qpdf/diff-ignore-ID-version
@@ -1,6 +1,6 @@
 #!/bin/sh
-lines=$(expr 0 + $(diff "$1" "$2" | egrep '^[<>]' | \
-        egrep -v '/ID' | egrep -v '%PDF-' | wc -l))
+lines=$(expr 0 + $(diff "$1" "$2" | grep -E '^[<>]' | \
+        grep -E -v '/ID' | grep -E -v '%PDF-' | wc -l))
 if [ "$lines" = "0" ]; then
    echo okay
 else
