diff -urN ../qpdf-4.0.1.orig/qpdf/qpdf.cc ./qpdf/qpdf.cc
--- ../qpdf-4.0.1.orig/qpdf/qpdf.cc	2013-01-17 14:51:04.000000000 +0000
+++ ./qpdf/qpdf.cc	2013-03-04 21:07:56.361335897 +0000
@@ -1574,7 +1574,7 @@
                 // This prevents those objects from being preserved by
                 // being referred to from other places, such as the
                 // outlines dictionary.
-                for (int pageno = 0; pageno < (int)orig_pages.size(); ++pageno)
+                for (size_t pageno = 0; pageno < orig_pages.size(); ++pageno)
                 {
                     if (selected_from_orig.count(pageno) == 0)
                     {
